src/main/java/com/epam/tc/hw7/pages/JdiMetalsAndColorsPage.java

```diff
@@ -3,57 +3,117 @@
-import com.epam.jdi.light.ui.html.elements.complex.DataListOptions;
 import com.epam.jdi.light.ui.html.elements.complex.RadioButtons;
+import com.epam.tc.hw7.entities.Data;
+import java.util.List;
+import org.hamcrest.Matchers;
 
 public class JdiMetalsAndColorsPage extends WebPage {
 
-    @Css("#odds-selector")
+    @UI("[name=custom_radio_odd]")
     public RadioButtons summaryFirstLine;
 
-    @Css("#even-selector")
+    @UI("[name=custom_radio_even]")
     public RadioButtons summarySecondLine;
 
-    @Css("p[class=checkbox]")
+    @UI("input[type=checkbox]")
     public Checklist elementsSettings;
 
-    @JDropdown(root = "div[ui=dropdown]",
-               value = ".filter-option",
+    @JDropdown(root = ".colors",
+               value = ".dropdown-value",
                list = "li",
                expand = ".caret")
-    public DataListOptions  colorsSettings;
+    public Dropdown colorsSettings;
 
-    @JDropdown(root = "div[ui=combobox]",
+    @JDropdown(root = ".metals",
+               value = ".dropdown-value",
                list = "li",
-               value = ".filter-option",
                expand = ".caret")
-    private DataListOptions  metals;
+    private Dropdown metals;
 
-    @JDropdown(root = "div[ui=droplist]",
+    @JDropdown(root = "#salad-dropdown",
+               value = ".dropdown-value",
                list = "li",
-               value = ".filter-option",
                expand = ".caret")
-    private DataListOptions vegetablesSettings;
+    private Dropdown vegetablesSettings;
 
-    @Css("#submit-button")
+    @UI("#submit-button")
     public Button submitButton;
 
-    @Css("li[class=summ-res]")
+    @UI(".summ-res")
     public Text summaryResult;
 
-    @Css("li[class=elem-res]")
+    @UI(".elem-res")
     public Text elementsResult;
 
-    @Css("li[class=col-res]")
+    @UI(".col-res")
     public Text colorResult;
 
-    @Css("li[class=met-res]")
+    @UI(".met-res")
     public Text metalResult;
 
-    @Css("li[class=sal-res]")
+    @UI(".sal-res")
     public Text vegetablesResult;
+
+
+
+    public void chooseSummary(List<Integer> summary) {
+        summaryFirstLine.select(summary.get(0).toString());
+        summarySecondLine.select(summary.get(1).toString());
+    }
+
+    public void chooseElements(List<String> elements) {
+        for (String element : elements) {
+            elementsSettings.select(element);
+        }
+    }
+
+    public void chooseColor(String colorsName) {
+        colorsSettings.setValue(colorsName);
+    }
+
+    public void chooseMetal(String metalsName) {
+        metals.select(metalsName);
+    }
+
+    public void chooseVegetables(List<String> vegetables) {
+        vegetablesSettings.select("Vegetables");
+        for (String vegetable : vegetables) {
+            vegetablesSettings.select(vegetable);
+        }
+    }
+
+    public void submitFormMetalsColors() {
+        submitButton.click();
+    }
+
+    public void checkResultLog(Data dataFromJson) {
```

Comment:
Very difficult to understand this function. Why you cann't just read text from Page and compare with the string from the data?

Reply:
changed